### PR TITLE
Handle image load failure

### LINKED ISSUE #87: Some images fail to load
Screen Shot 2021-09-22 at 8 23 34 PM

Some images within an article card fail to load. When this happens, the alt text and a placeholder icon gets substituted for the original image. Whenever an image fails to load, we should prevent the alt text and placeholder from displaying. I'm not sure if we can prevent this, though. It may have something to do with [weserv](https://images.weserv.nl/), an image caching service we use

### PR DESCRIPTION
**Pull request by mcflav**
Fixes #87 

image

### CODE REVIEW

File: src/features/globalReadings/components/ListItem.tsx [hunk 1]
```diff
@@ -109,8 +109,15 @@ const ListItem: React.FunctionComponent<ListItemProps> = ({
             <img
               loading="lazy"
               src={newReadingImage}
+              onError={(e) => {
```
Comment (aubundy) [inline src/features/globalReadings/components/ListItem.tsx:112]: Can you move this anonymous function inside `imageOnErrorHandler()` and then move that function to be within `ListItem` below `getImageDimensions()`?

File: src/features/globalReadings/components/ListItem.tsx [hunk 2]
```diff
@@ -109,8 +109,15 @@ const ListItem: React.FunctionComponent<ListItemProps> = ({
...
+                let source = e.target as HTMLImageElement;
+                source.onerror = null;
+                source.src = "";
+                source.remove();
+              }}
+              //onError={imageOnErrorHandler}
               onLoad={measure}
-              alt="Article"
+              // alt='Article'
```
Comment (aubundy) [inline src/features/globalReadings/components/ListItem.tsx:120]: Please remove this comment

File: src/features/globalReadings/components/ListItem.tsx [hunk 3]
```diff
@@ -109,8 +109,15 @@ const ListItem: React.FunctionComponent<ListItemProps> = ({
             <img
               loading="lazy"
               src={newReadingImage}
+              onError={(e) => {
+                let source = e.target as HTMLImageElement;
+                source.onerror = null;
+                source.src = "";
+                source.remove();
+              }}
+              //onError={imageOnErrorHandler}
```
Comment (aubundy) [inline src/features/globalReadings/components/ListItem.tsx:118]: Please remove this commented out code

File: src/features/globalReadings/components/ListItem.tsx [hunk 4]
```diff
@@ -169,6 +176,13 @@ function mapStateToProps(state: RootState, ownProps: OwnProps) {
 
 const connector = connect(mapStateToProps);
 
+//if image link is broken during loading the component do not show broken link or placeholder.
+const imageOnErrorHandler = (
```
Comment (aubundy) [inline src/features/globalReadings/components/ListItem.tsx:180]: See note on this function above

**Review by aubundy — CHANGES_REQUESTED** [review]
This code works for me! I just added a few changes to move the logic around

**Comment by aubundy** [comment]
[Image of working code](https://github.com/zero-to-mastery/breads-client/issues/87#issuecomment-943965265)

**Comment by mcflav** [comment]
This has been completed. Please review, thanks!

________________________________

**Pull request closed without merging**